js/ui/discoveryFeedButton.js

```diff
@@ -54,21 +50,54 @@ function maybeCreateButton() {
  */
 const DiscoveryFeedButton = new Lang.Class({
     Name: 'DiscoveryFeedButton',
-    Extends: St.Button,
+    Extends: St.BoxLayout,
 
     _init: function() {
-        this.parent({ name: 'discovery-feed',
-                      child: new St.Icon({ icon_name: 'go-down-symbolic',
-                                           icon_size: 16,
-                                           style_class: 'discovery-feed-icon ' }),
-                      style_class: 'discovery-feed-button',
+        this.parent({ vertical: true,
                       visible: _primaryMonitorWidthPassesThreshold() });
 
+        this._bar = new St.Button({ name: 'discovery-feed-bar',
+                                    child: new St.Icon({ style_class: 'discovery-feed-bar-icon' }),
+                                    style_class: 'discovery-feed-bar',
+                                    reactive: true  });
+        this.add(this._bar);
+
+        this._tile = new St.Button({ name: 'discovery-feed-tile',
+                                     child: new St.Icon({ style_class: 'discovery-feed-tile-icon' }),
+                                     style_class: 'discovery-feed-tile' });
+        this.add(this._tile, { x_fill: false,
+                               x_align: St.Align.MIDDLE,
+                               expand: true });
+
+        this._bar.connect('event', Lang.bind(this, this._onEvent));
+        this._tile.connect('event', Lang.bind(this, this._onEvent));
+
+        this._bar.connect('notify::hover', Lang.bind(this, this._onHoverChanged));
+        this._tile.connect('notify::hover', Lang.bind(this, this._onHoverChanged));
+
         Main.layoutManager.connect('monitors-changed', Lang.bind(this, function() {
             this.visible = _primaryMonitorWidthPassesThreshold();
         }));
     },
 
+    _onEvent: function(actor, event) {
+        if (event.type() == Clutter.EventType.TOUCH_BEGIN ||
+            event.type() == Clutter.EventType.BUTTON_PRESS)
```

Comment:
I think you may want to use `TOUCH_END` and `BUTTON_RELEASE` here - a "click" or "touch" is only fully completed once one of those two things happens.

Reply:
We have both behaviours in the shell. There was a specific request to make this on button_press see https://phabricator.endlessm.com/T19354#516229. This is the same behaviour as with the Endless button in the taskbar: https://github.com/endlessm/gnome-shell/blob/master/js/ui/endlessButton.js#L79